pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
+	LdapGroupsHeaderName string
```

Comment:
task: LDAP is only one of many possible content values of the headers. I don't think this plugin should tightly couple to LDAP even if only by name.

Reply:
question: related to this, should this plugin be related to authz in any respect? or simply a generic header validation component

Reply:
fixed